### ISSUE TITLE
Raiden node crashes regularly after running for some time

### ISSUE THREAD

**Issue by agatsoh** [raiden-network/raiden#6032]
## Problem Definition
While Testing raiden develop branch on `dbe3f21f54fb36f66b03232d182b68d6c4960861` commit (24th march 2020), I encountered that raiden nodes were getting shut randomly after some time of starting. 
Not really sure whether it is because of the PFS, the matrix or an unstable internet connection. 

The logs to all the nodes can be found here
https://drive.google.com/open?id=1QRbPAUNMjpsCUVLsIOI4huSBEq_hy6ON

**Comment by hackaugusto** [comment]
@agatsoh could you please re-upload the log of the failing node compressed? I want to double check if hte files are corrupted.

**Comment by Dominik1999** [comment]
@agatsoh can you consistently reproduce that?

**Comment by agatsoh** [comment]
@hackaugusto I cannot consistently reproduce it, the logs of 3 days that i have uploaded. I observed nodes abruptly closing on 24th March and 26th March, on 25th it it seemed to run relatively ok. But overall the experience is not particularly good regarding the uptime of nodes.

**Comment by Dominik1999** [comment]
- Augusto saw some comms errors with Matrix
- Some logs end abruptly (Sudden Node Death 2) 
- check Sohans system logs?

**Comment by agatsoh** [comment]
The logs of the same kind of crashing behaviour i observed in the 0.200.0 RC3
@Dominik1999 @hackaugusto 
https://drive.google.com/file/d/1E1raG5VXhRolti912IV47Kwj4IFCCSKe/view?usp=sharing

**Comment by hackaugusto** [comment]
Notes:

The logs for the day `24` show the nodes exited with a `SIGPIPE`, which is expected behavior.

**Comment by karlb** [comment]
I also had a few cases of shutting down due to SIGPIPE. In which cases is this expected? Is it bad internet connection, restarting servers, etc?
In my cases the cli stayed completely silent, which is probably a UI problem if this is considered to be expected behavior.

**Comment by hackaugusto** [comment]
I was thinking about it, perhaps the node was executed in a remote server through a `ssh` connection? or in a terminal multiplexer, e.g. `screen` or `tmux` and the window was detached? AFAIK `SIGPIPE` are used when one of the programs in a pipe exits, closing the `stdint`/`stdout` (e.g. `cat f | head`, `cat` will get a `SIGPIPE` since `head` exists after 10 lines).

ref: https://www.gnu.org/software/libc/manual/html_node/Operation-Error-Signals.html#Operation-Error-Signals

**Comment by karlb** [comment]
I do run raiden in tmux, but there was no occasion where tmux closed std*. I just got the SIGPIPE five times during the last 30 mins. Everytime it happened when reloading the WebUI after staring the raiden node. But I can't reproduce it intentionally, yet.

**Comment by hackaugusto** [comment]
@karlb if you can reproduce it, then try adding a [`print_tb`](https://docs.python.org/3/library/traceback.html#traceback.print_tb) for the [`frame` on the signal handler](https://github.com/raiden-network/raiden/blob/68b2cb0f185a38728652c5ab595bbd1407bcf036/raiden/ui/runners.py#L55-L56). I have no idea which frame will be reported, but it may give some insight.

Just to make sure, the process is being stopped and then a new one started, right? otherwise we could have some leaked sockets laying around

**Comment by hackaugusto** [comment]
@karlb managed to get a stack trace:

```
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/baseserver.py", line 34, in _handle_and_close_when_done                                                                     
    return handle(*args_tuple)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 1514, in handle                                                                                            
    handler.handle()
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 457, in handle                                                                                             
    result = self.handle_one_request()
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 678, in handle_one_request                                                                                 
    self.handle_one_response()
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 968, in handle_one_response                                                                                
    self.run_application()
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 916, in run_application                                                                                    
    self.process_result()
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 902, in process_result                                                                                     
    self.write(data)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 749, in write                                                                                              
    self._write_with_headers(data)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 770, in _write_with_headers                                                                                
    self._write(data)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 734, in _write                                                                                             
    self._sendall(data)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/pywsgi.py", line 708, in _sendall                                                                                           
    self.socket.sendall(data)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/_socket3.py", line 521, in sendall                                                                                          
    return _socketcommon._sendall(self, data_memory, flags)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/_socketcommon.py", line 380, in _sendall                                                                                    
    timeleft = __send_chunk(socket, chunk, flags, timeleft, end)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/_socketcommon.py", line 309, in __send_chunk                                                                                
    data_sent += socket.send(chunk, flags)
  File "/home/karl/tools/venv/lib/python3.7/site-packages/gevent/_socket3.py", line 502, in send                                                                                             
    return self._sock.send(data, flags)
  File "/home/karl/raiden/raiden/ui/runners.py", line 58, in sig_set
    traceback.print_stack()
```

Installing a handler for `SIGPIPE` was not a good idea. The fix #6044 removes the handler and adds an extensive explanation for the existing behavior.

Note: The error reported happened whenever the webui was refreshed. When the ui is refreshed the client socket is closed, but the webserver doesn't know about it until the reponse is being written to the server socket. When the server tries to write the http reponse with the body the system realizes the socket is closed, and the `SIGPIPE` is emitted. The default behavior in the interpreter runtime is to *not* have a handler, instead to configure the syscall to return an error code, which in turn is converted by the runtime to an exception. This works fine for `raiden | head` and `socket`s since each call site can handle the runtime error individually. Installing the handler like we had, will erroneously stop the client when a client closed a socket, because there is no way to differentiate if the signal was created because of a write to a closed socket or a write to a closed stdout.

**Comment by Dominik1999** [comment]
nice!
